### PR TITLE
ebook2cw: fix darwin build

### PR DESCRIPTION
**Pull request by stephank**
###### Motivation for this change

ZHF: #122042
cc @NixOS/nixos-release-managers

Hydra failure: https://hydra.nixos.org/build/141984333/log

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/ebook2cw/Makefile.patch [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+index 005d60d..0718f7e 100644
 --- a/Makefile
 +++ b/Makefile
-@@ -41,6 +41,7 @@ install:
+@@ -24,23 +24,24 @@ endif
+ all: ebook2cw
+ 
+ ebook2cw: ebook2cw.c codetables.h
+-	gcc ebook2cw.c -pedantic -Wall -lm $(LDFLAGS) $(CFLAGS) -o ebook2cw
++	$(CC) ebook2cw.c -pedantic -Wall -lm $(LDFLAGS) $(CFLAGS) -o ebook2cw
```
Comment (jonringer) [inline pkgs/applications/radio/ebook2cw/Makefile.patch:10]: do you mind upstreaming this if possible? doesn't seem to be a nix-specific patch

**Comment by risicle** [comment]
`nixpkgs-review` happy for me, macos 10.15, nixos x86_64.

File: pkgs/applications/radio/ebook2cw/Makefile.patch (hunk 1 above)
Comment (stephank) [reply pkgs/applications/radio/ebook2cw/Makefile.patch:10]: The issue tracker appears to be private. I guess I'll mail them to make them aware of our patches. 😕

Comment (stephank) [reply pkgs/applications/radio/ebook2cw/Makefile.patch:10]: Good thing is: upstream is super responsive and merged it within 15 minutes. :)

**Comment by stephank** [comment]
I added an upgrade to 0.8.4, because that allows us to fetchpatch from upstream. (It doesn't apply to 0.8.3.)

**Pull request merged**